### ISSUE TITLE
Plugin seems to crash Ardour after audio export

### ISSUE THREAD

**Issue by WilhelmKuntner** [x42/zconvo.lv2#2]
Hi Robin, 

I use the x42 IR Convolver plugin in Ardour and it seems to be the cause of crashes right after the audio export has finished. The exported audio sound like expected, that is, audio export works and so does the plugin, in terms of output. 

I run Ardour without Jack, only with ALSA and I've tried to uninstall Pulseaudio, but it still crashes. The Present Convolver Plugin doesn't crash Ardour. 

Please let me know if you need/want any kind of info or backtrace. 

A huge thank you for the overwhelming plugins; the Preset Convolver with the IR descriptions is extremely nice and so are all the other plugins that you provided. Let me know if and how I can help, Wilhelm.

**Comment by x42** [comment]
Odd that the preset convolver does not crash, that's effectively the same code just with a different IR.

If you can get a backtrace that would be helpful, alternatively could you zip up the session that produces the bug and send it to me (robin@gareus.org) so that I can reproduce the problem?

**Comment by WilhelmKuntner** [comment]
I did (backtrace), but I'm not sure about the "zip up" part of your message, please let me know in case you need more than the session file.

**Comment by x42** [comment]
OK.After export completes Ardour resets plugins (de + re-activate). The crash happens there

```
Thread 33 (Thread 0x7fffd1445900 (LWP 28875)):
#0 tcache_get (tc_idx=9) at malloc.c:2934
0000001 __GI___libc_malloc (bytes=160) at malloc.c:3042
#2 0x00007fffb813df6d in ?? () from /home/wilhelm/.lv2/zeroconvo.lv2/zeroconvolv.so
#3 0x00007fffb813de95 in ?? () from /home/wilhelm/.lv2/zeroconvo.lv2/zeroconvolv.so
[..]
#15 0x00007fffb813c4fd in ?? () from /home/wilhelm/.lv2/zeroconvo.lv2/zeroconvolv.so
0000016 0x00007fffb8136344 in ?? () from /home/wilhelm/.lv2/zeroconvo.lv2/zeroconvolv.so
#17 0x00007ffff765372a in ARDOUR::LV2Plugin::activate() () from /opt/Ardour-6.9.0/lib/libardour.so.3
[..]
```

I don't have an explanation why this would crash in `malloc`. Does your system have a limited amount of locked memory?
What is the output of `prlimit --memlock`  and/or `ulimit -a`?

Have you compiled the plugin yourself? If so a debug version of the plugin would be useful.
`make OPTIMIZATIONS="-g" STRIP=/bin/true`
If not, I could  do a debug build for you in the next days. Linux, 64 bit Intel?

**Comment by x42** [comment]
A debug build of the plugin is available from https://x42-plugins.com/tmp/x42-zconvolver-v0.5.2-x86_64-debug.tar.gz

```bash
# make a backup of the current plugin
mv -vi ~/.lv2/zeroconvo.lv2 /tmp/
# download the debug version of the plugin and install it
cd /tmp
wget https://x42-plugins.com/tmp/x42-zconvolver-v0.5.2-x86_64-debug.tar.gz
tar xvf x42-zconvolver-v0.5.2-x86_64-debug.tar.gz
mv -vi x42-zconvolver/zeroconvo.lv2 ~/.lv2/
```

Now re-create the crash and get a backtrace of the crash after export

and then

```bash
# restore the original, optimized plugin
rm -rf ~/.lv2/zeroconvo.lv2
mv -vi /tmp/zeroconvo.lv2 ~/.lv2/
```

**Comment by WilhelmKuntner** [comment]
(Sorry for double posting:)

$ prlimit --memlock
RESOURCE DESCRIPTION                             SOFT      HARD UNITS
MEMLOCK  max locked-in-memory address space unlimited unlimited bytes

$ ulimit -a
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
scheduling priority             (-e) 0
file size               (blocks, -f) unlimited
pending signals                 (-i) 47401
max locked memory       (kbytes, -l) unlimited
max memory size         (kbytes, -m) unlimited
open files                      (-n) 1024
pipe size            (512 bytes, -p) 8
POSIX message queues     (bytes, -q) 819200
real-time priority              (-r) 90
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 47401
virtual memory          (kbytes, -v) unlimited
file locks                      (-x) unlimited

**Comment by WilhelmKuntner** [comment]
Please find attached the backtrace that I generated according to your instructions, many thanks. 

[gdb.txt](https://github.com/x42/zconvo.lv2/files/8176750/gdb.txt)

**Comment by x42** [comment]
Thanks. potential fix in 2ba6328a4b15c.

A binary to test: https://x42-plugins.com/tmp/x42-zconvolver-v0.5.2-1-x86_64-debug.tar.gz

**Comment by x42** [comment]
Diyou you have the chance to test it? Does the fix work?

**Comment by WilhelmKuntner** [comment]
The fix has worked so far as I have exported audio without any crashes several times; I tried both the real-time and the "fast" audio export.

I downloaded the fixed file from the link that you sent me, compiled it, and substituted the debug version. Compilation returned some errors (I have no idea what) but the file was build and works. Good for me. Thank you very much!

Let me know if you want to do more tests.
